OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift

```diff
@@ -10,118 +10,133 @@ import Foundation
         
-        if encodedData.count < 19 {
+        if encodedData.count < 21 {
             throw MessageBlockError.notEnoughData
         }
         
-        guard let reservoirStatus = ReservoirStatus(rawValue: encodedData[1]) else {
-            throw MessageError.unknownValue(value: encodedData[1], typeDescription: "ReservoirStatus")
+        guard PodProgressStatus(rawValue: encodedData[1]) != nil else {
+            throw MessageError.unknownValue(value: encodedData[1], typeDescription: "PodProgressStatus")
         }
-        self.reservoirStatus = reservoirStatus
+        self.podProgressStatus = PodProgressStatus(rawValue: encodedData[1])!
         
         self.deliveryType = DeliveryType(rawValue: encodedData[2] & 0xf)
         
         self.insulinNotDelivered = podPulseSize * Double((Int(encodedData[3] & 0x3) << 8) | Int(encodedData[4]))
         
         self.podMessageCounter = encodedData[5]
-        self.unknownPageCode = Double(Int(encodedData[6]) | Int(encodedData[7]))
+        self.unknownPageCode = encodedData[6...7]
         
-        self.originalLoggedFaultEvent = FaultEventCode(rawValue: encodedData[8])
+        self.currentStatus = FaultEventCode(rawValue: encodedData[8])
         
-        self.faultEventTimeSinceActivation = TimeInterval(minutes: Double((Int(encodedData[9] & 0b1) << 8) + Int(encodedData[10])))
+        self.faultEventTimeSinceActivation = TimeInterval(minutes: Double(encodedData[9...10].toBigEndian(UInt16.self)))
         
-        self.insulinRemaining = podPulseSize * Double((Int(encodedData[11] & 0x3) << 8) | Int(encodedData[12]))
+        let resHighBits = Int(encodedData[11] & 0x03) << 6
+        let resLowBits = Int(encodedData[12] >> 2)
+        let reservoirValue = round(Double((resHighBits + resLowBits) * 50)/255)
+        if reservoirValue < StatusResponse.maximumReservoirReading {
+            reservoirLevel = "\(reservoirValue) U"
+        } else {
+            reservoirLevel = ">50 U"
+        }
         
-        self.timeActive = TimeInterval(minutes: Double((Int(encodedData[13] & 0b1) << 8) + Int(encodedData[14])))
+        self.timeActive = TimeInterval(minutes: Double(encodedData[13...14].toBigEndian(UInt16.self)))
         
-        self.secondaryLoggedFaultEvent = FaultEventCode(rawValue: encodedData[15])
+        self.previousStatus = FaultEventCode(rawValue: encodedData[15])
         
         self.logEventError = encodedData[16] == 2
 
-        self.insulinStateTableCorruption = encodedData[17] & 0b10000000 != 0
-        
-        guard let reservoirStatusAtFirstLoggedFaultEventType = ReservoirStatus(rawValue: encodedData[17] & 0xF) else {
-            throw MessageError.unknownValue(value: encodedData[17] & 0xF, typeDescription: "ReservoirStatus")
+        guard let logEventErrorType = LogEventErrorType(rawValue:encodedData[17] >> 4) else {
+            throw MessageError.unknownValue(value: encodedData[17] >> 4, typeDescription: "LogEventErrorType")
```

Comment:
I think this should maybe not throw an exception. I'm guessing we are likely to encounter new values for this, so we should just record it as an unknown type, like we do for FaultCodes.  If we're very confident of not receiving new, unknown values here, then we can throw.